### DISCUSSION TITLE
ProviderObserver does not return provider name.

### DISCUSSION THREAD

**Discussion by subhasundardass** [rrousselGit/riverpod#1458] [General]
class RiverpodLogger extends ProviderObserver {
  @override
  void didUpdateProvider(
    ProviderBase provider,
    Object? previousValue,
    Object? newValue,
    ProviderContainer container,
  ) {
    if (kDebugMode) {
      if (newValue is StateController<int>) {
        print('''
        ======
          "provider": "${provider.name ?? provider.runtimeType}",
          "newValue": "${newValue.state}"
        ======''');
      }
    }
  }
}

#------------------------
provider.name always return null. How do I get provider name? please help

**Comment by rrousselGit** [comment]
You have to define one on your provider:

```dart
Provider(..., name: 'Hello')
```

**Reply by subhasundardass** [reply]
Many many thanks, You have done great job, Just leave Getx and started using Riverpod.
